server.js

```diff
@@ -16,4 +16,178 @@ MongoClient.connect('mongodb://localhost', { useNewUrlParser: true }).then(conne
   });
 }).catch(error => {
   console.log('ERROR:', error);
-});
\ No newline at end of file
+});
+
+
+/*
+Request Type: GET
+Endpoint: /api/videos/:id
+Params: Wikipedia page id
+Returns: list of videos
+Ex:
+		{
+    "videos": [
+        {
+            "_id": "5ca41e15346f29bdff6dd083",
+            "wikiPageId": 76894,
+            "sectionIdx": 3,
+            "created": "2016-08-15T00:00:00.000Z",
+            "ytId": "Bxpu6tbFCsI",
+            "title": "Emu War - OverSimplified (Mini-Wars #4)",
+            "upvotes": 3,
+            "downvotes": 0
+        },
+...
+*/
+
+app.get('/api/videos/:id', (req,res) => {
+  let wikiId = parseInt(req.params.id,10); //converting from string to number
+  db.collection('videos').find({wikiPageId: wikiId}).toArray().then(pageVideos => { //finds all video objects pertaining to a page, converts it to an array 
+    res.json({videos: pageVideos}) //returns list of videos of that pageId, empty if either the pageId doesn't exist or there are no videos currently for it
+  }).catch(error => {
+    console.log(error);
+    res.status(500).json({ message: `Error in parsing GET request to /api/videos/${wikiId}: ${error}`});
+  });
+});
+
+
+/*
+Request Type: POST
+Endpoint: /api/videos
+Params: video object
+If no video object present with same values, create new video document and return it with new properties upvote and downvote, and success = true
+Else, return success = false
+Returns: id of updated document
+Request Body Ex:
+		{
+			"video":{
+				"wikiPageId": 76895,
+				"sectionIdx": 7,
+				"ytId": "abcd"
+			}
+}
+Response Body Ex:
+Upon success:
+{
+    "success": true,
+    "video": {
+        "wikiPageId": 76895,
+        "sectionIdx": 7,
+        "ytId": "abcd",
+        "upvotes": 0,
+        "downvotes": 0
+    }
+}
+			
+Upon failure:
+{
+    "success": false
+}
+*/
+
+app.post('/api/videos', (req,res) => {
+  let newVideo = req.body.video;
+
+  //first we search the database to see if the video object already exists
+  db.collection('videos').find({wikiPageId: newVideo.wikiPageId, sectionIdx: newVideo.sectionIdx, ytId: newVideo.ytId}).count().then(count => {
+    if(count != 0){ //if it already exists, we don't add it
+      res.json({
+        success: false //if we don't add it, we set the response body success property to false
+      })
+    }
+    else{ //if it does exist
+      // Adding/reassigning/initializing properties of video object to be added to our collection
+      newVideo.created = new Date();
+      newVideo.upvotes = 0;
+      newVideo.downvotes = 0;
+      db.collection('videos').insertOne(newVideo).then(result => //Adding it to our collection
+        db.collection('videos').find({_id: result.insertedId}).limit(1).next() //finding it again to return it as part of the JSON
+      ).then(newVideo => {
+        res.json({
+          success: true, //if we do add it, we set the response body success property to true
+          video: {

```

Comment:
Could this just be like
```suggestion
          video: newVideo
```